File: app/src/main/java/com/kickstart/application/di/module/ApplicationModule.java
package com.kickstart.application.di.module;

import android.content.Context;

import com.kickstart.application.AndroidApplication;
import com.kickstart.application.di.component.ActivityComponent;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Module which provides objects which will live throughout the application lifecycle.
 */
@Module(subcomponents = {ActivityComponent.class})
public class ApplicationModule {
    private final AndroidApplication mApplication;

    public ApplicationModule(AndroidApplication application) {
        this.mApplication = application;
    }

    @Provides @Singleton
    Context provideApplicationContext() {
        return this.mApplication;
    }
}

File: app/src/main/java/com/kickstart/railway/di/module/ApplicationModule.kt
package com.kickstart.railway.di.module

import android.content.Context

import com.kickstart.railway.AndroidApplication
import com.kickstart.railway.di.component.ActivityComponent

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

/**
 * Module which provides objects which will live throughout the application lifecycle.
 */
@Module(subcomponents = arrayOf(ActivityComponent::class))
class ApplicationModule(private val mApplication: AndroidApplication) {

    @Provides
    @Singleton
    internal fun provideApplicationContext(): Context {
        return this.mApplication
    }
}
